climada/hazard/centroids/centr.py

```diff
@@ -555,60 +555,48 @@ def union(self, *others):
         ------
             ValueError
         """
-        cent_list = list(others)
-        if self.lat.any() or self.meta:
-            cent_list.insert(0, self)
-
-        if np.all([cent_list[0].equal(cent) for cent in cent_list[1:]]):
-            return copy.deepcopy(cent_list[-1])
-
-        #Set attributes to centroids if missing in one but defined in other
-        is_area_pixel = np.any([cent.area_pixel.any() for cent in cent_list]) \
-            and not np.all([cent.area_pixel.any() for cent in cent_list])
-        is_dist_coast = np.any([cent.dist_coast.any() for cent in cent_list]) \
-            and not np.all([cent.dist_coast.any() for cent in cent_list])
-        is_on_land = np.any([len(cent.on_land) > 0 for cent in cent_list]) \
-            and not np.all([len(cent.on_land) > 0 for cent in cent_list])
-        is_region_id = np.any([cent.region_id.any() for cent in cent_list]) \
-            and not np.all([cent.region_id.any() for cent in cent_list])
-        is_elevation = np.any([cent.elevation.any() for cent in cent_list]) \
-            and not np.all([cent.elevation.any() for cent in cent_list])
+        # restrict to non-empty centroids
+        cent_list = [c for c in (self,) + others if c.size > 0 or c.meta]
+        if len(cent_list) == 0 or len(others) == 0:
+            return copy.deepcopy(self)
+        self = cent_list[0]
```

Comment:
That looks like a very strange line. Replace self with and `=` ?

Reply:
I expected that you wouldn't like this, so I won't argue, but just changed it in my latest commit :smile: